ethportal-api/src/types/node_id.rs

```diff
@@ -9,53 +8,25 @@ use super::distance::{Metric, XorMetric};
 
 type RawNodeId = [u8; 32];
 
-/// Discv5 NodeId
-// TODO: Wrap this type over discv5::NodeId type
-#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
-pub struct NodeId(#[serde(with = "SerHex::<StrictPfx>")] pub RawNodeId);
-
-impl NodeId {
-    pub fn random() -> Self {
-        Self(rand::random())
-    }
-
-    pub fn raw(self) -> RawNodeId {
-        *self.deref()
-    }
-
-    /// Generate random NodeId based on bucket index target and a local node id.
-    /// First we generate a random distance metric with leading zeroes based on the target bucket.
-    /// Then we XOR the result distance with the local NodeId to get the random target NodeId
-    // TODO: We should be able to make this generic over a `Metric`.
-    pub fn generate_random_node_id(target_bucket_idx: u8, local_node_id: NodeId) -> NodeId {
-        let distance_leading_zeroes = 255 - target_bucket_idx;
-        let random_distance = crate::utils::bytes::random_32byte_array(distance_leading_zeroes);
-
-        let raw_node_id = XorMetric::distance(&local_node_id.raw(), &random_distance);
+/// Generate random NodeId based on bucket index target and a local node id.
+/// First we generate a random distance metric with leading zeroes based on the target bucket.
+/// Then we XOR the result distance with the local NodeId to get the random target NodeId
+// TODO: We should be able to make this generic over a `Metric`.
+pub fn generate_random_node_id(target_bucket_idx: u8, local_node_id: EnrNodeId) -> EnrNodeId {
+    let distance_leading_zeroes = 255 - target_bucket_idx;
+    let random_distance = crate::utils::bytes::random_32byte_array(distance_leading_zeroes);
 
-        let raw_bytes = raw_node_id.big_endian();
+    let raw_node_id = XorMetric::distance(&local_node_id.raw(), &random_distance);
 
-        raw_bytes.into()
-    }
-}
-
-impl From<RawNodeId> for NodeId {
-    fn from(item: RawNodeId) -> Self {
-        NodeId(item)
-    }
-}
+    let raw_bytes = raw_node_id.big_endian();
 
-impl From<EnrNodeId> for NodeId {
-    fn from(value: EnrNodeId) -> Self {
-        NodeId(value.raw())
-    }
+    raw_bytes.into()
 }
 
-impl From<NodeId> for EnrNodeId {
-    fn from(val: NodeId) -> Self {
-        EnrNodeId::new(val.deref())
-    }
-}
+/// Discv5 NodeId
+// TODO: Wrap this type over discv5::NodeId type
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct NodeId(#[serde(with = "SerHex::<StrictPfx>")] pub RawNodeId);
```

Comment:
We don't need this structure anymore, we can use `discv5::enr::NodeId`.

Reply:
https://github.com/sigp/enr/pull/34 here I list why we need it only for tracing. Without ``#[serde(with = "SerHex::<StrictPfx>")`` ``json!(Hashmap<NodeId, xyz>) will panic. This is used in the query tracing code.

If the concensus is to just wait till the next discv5 till this is resolved, that is fine. I just thought we could do most of the transition.

Reply:
I updated my write up with the information as well, https://github.com/ethereum/trin/pull/669#issue-1669551007 . It is unfortunate.

Reply:
I see, ok, no need to wait for IMO but we need to create an issue after this is merged.